### ISSUE TITLE
dex: value breaker does not account for `delta_2`

### ISSUE THREAD

**Issue by erwanor** [penumbra-zone/penumbra#3890]
In #3690, we added a value circuit breaker to the dex component. The value circuit breaker maintains an aggregate `Balance` for the value held in liquidity positions. After handling a batch swap, we check if the batched output exceed this allowance and halt if so.

Zellic was able to trigger this halt by sending an innocuous-but-irrational swap (with both $\Delta_1, \Delta_2 \neq 0$). We should investigate what is the bug, and remediate it.

Something suspicious is that the value breaker assert does not factor in input amounts:

```rust
        // Check that the output data doesn't exceed the ValueCircuitBreaker's quantities
        // (i.e. we didn't outflow more value than existed within liquidity positions).
        let available_asset_1 = value_circuit_breaker.available(trading_pair.asset_1());
        let available_asset_2 = value_circuit_breaker.available(trading_pair.asset_2());
        assert!(
            output_data.lambda_1 <= available_asset_1.amount,
            "asset 1 outflow exceeds available balance"
        );
        assert!(
            output_data.lambda_2 <= available_asset_2.amount,
            "asset 2 outflow exceeds available balance"
        );
```

**Comment by erwanor** [comment]
We should take this as an opportunity to simplify value tracking and shed some IO from `put_position`:

```
there are two places that value flows into the dex and two places it flows out, right?

IN:
swap inputs
opened liquidity positions
OUT:
swap outputs
withdrawn positions

if we check those places we are guaranteed to check that inflows > outflows
```

(full conversation: https://discord.com/channels/824484045370818580/938328380917559326/1216870835894358156)

**Comment by erwanor** [comment]
Fixed in #4086
